presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -582,6 +591,16 @@ public void grantTablePrivileges(String databaseName, String tableName, String g
     public void revokeTablePrivileges(String databaseName, String tableName, String grantee, Set<PrivilegeGrantInfo> privilegeGrantInfoSet)
     {
         try {
+            Set<HivePrivilegeInfo> existingPrivileges = getTablePrivileges(grantee, databaseName, tableName);
+
+            Set<PrivilegeGrantInfo> privilegesToRevoke = privilegeGrantInfoSet.stream()
+                    .filter(requestedPrivilege -> existingPrivileges.containsAll(parsePrivilege(requestedPrivilege)))
```

Comment:
And what about "ALL". Cannot it be used as `requestedPrivilege`?
It is parsed to set with multiple values:

```
            case "ALL":
                return Arrays.asList(HivePrivilege.values()).stream()
                        .map(hivePrivilege -> new HivePrivilegeInfo(hivePrivilege, withGrantOption))
                        .collect(Collectors.toSet());
```